app/controllers/admin/companies_controller.rb

```diff
@@ -7,4 +7,11 @@ def destroy
     Company.find(params[:id]).destroy
     redirect_to admin_dashboard_index_path
   end
+
+  def update
+    company = Company.find(params[:id])
+    company.update(status: params[:company][:status])
+    company.save
```

Comment:
I don't have a conditional here (if save, redirect, else render :edit...:back) because the options for updating the status are hard coded, so I didn't think it was necessary, but do let me know if anyone disagrees.